File: src/main/java/xyz/mizarc/persistentitems/commands/PISubCommands/AddCommand.java
package xyz.mizarc.persistentitems.commands.PISubCommands;

import co.aikar.commands.BaseCommand;
import co.aikar.commands.annotation.CommandAlias;
import co.aikar.commands.annotation.CommandPermission;
import co.aikar.commands.annotation.Dependency;
import co.aikar.commands.annotation.Subcommand;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import xyz.mizarc.persistentitems.ItemConfigIO;

@CommandAlias("persistentitems|pitems|pi")
public class AddCommand extends BaseCommand {

    @Dependency
    ItemConfigIO itemConfig;

    @Subcommand("add")
    @CommandPermission("persistentitems.command.add")
    public void onAdd(Player player, String itemId, int slot) {
        ItemStack heldItemStack = player.getInventory().getItemInMainHand();
        itemConfig.addItem(itemId, heldItemStack, slot);
        player.sendMessage("Held item " + heldItemStack.getItemMeta().getDisplayName() + " has been added");
    }
}

File: src/main/kotlin/xyz/mizarc/persistentitems/commands/persistentitems/AddCommand.kt
package xyz.mizarc.persistentitems.commands.persistentitems

import co.aikar.commands.annotation.*
import org.bukkit.entity.Player
import xyz.mizarc.persistentitems.Item
import xyz.mizarc.persistentitems.commands.PersistentItemsCommand

@CommandAlias("persistentitems|pitems|pi")
class AddCommand : PersistentItemsCommand() {
    @Subcommand("add")
    @CommandPermission("persistentitems.command.add")
    @CommandCompletion("@nothing @nothing")
    @Syntax("<item> <slot>")
    fun onAdd(player: Player, name: String, slot: Int) {
        val heldItemStack = player.inventory.itemInMainHand
        itemRepo.add(Item(name, heldItemStack, slot))
        player.sendMessage("Held item ${heldItemStack.itemMeta.displayName()} has been added.")
    }
}
